src/registry/customerProfile/validationSchema.ts

```diff
@@ -0,0 +1,172 @@
+import type {
+  CustomerProfileComponentSchema,
+  DigitalAddress,
+  DigitalAddressType,
+} from '@open-formulieren/types';
+import type {IntlShape} from 'react-intl';
+import {defineMessage} from 'react-intl';
+import {z} from 'zod';
+
+import type {GetValidationSchema} from '@/registry/types';
+
+const USE_ONLY_ONCE_AND_AS_PREFERRED_CONSTRAINT_MESSAGE = defineMessage({
+  description:
+    'Validation error for customerProfile digital address useOnlyOnce and isNewPreferred that are both true',
+  defaultMessage: "The digital address cannot be marked as both 'one-time use' and 'preferred'.",
+});
+
+const DUPLICATE_DIGITAL_ADDRESS_TYPES_MESSAGE = defineMessage({
+  description:
+    'Validation error for customerProfile with multiple digital addresses for the same type',
+  defaultMessage: 'You cannot submit multiple digital addresses for the type {digitalAddressType}.',
+});
+
+const EMAIL_INVALID_MESSAGE = defineMessage({
+  description:
+    'Validation error for customerProfile digital address that does not match the email format',
+  defaultMessage: 'Invalid email address.',
+});
```

Comment:
So I'd expect the message for an invalid email to be identical if it's for the `email` or `customerProfile` component, which probably motivates defining this message string centrally and re-use it in both places, and give it a description that is decoupled from either of those components.

Otherwise in the translation catalog we need to ensure the same translations are used, and that's going to fail at some point, guaranteed.

`src/i18n/messages.ts` looks like a good candidate, where you use `defineMessage(s)` for the shared translations.